src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -1664,11 +1759,16 @@ class MavlinkStreamHomePosition : public MavlinkStream
 		return "HOME_POSITION";
 	}
 
-	uint8_t get_id()
+	static uint8_t get_id_static()
 	{
 		return MAVLINK_MSG_ID_HOME_POSITION;
 	}
 
+    uint8_t get_id()
```

Comment:
There is other code that depends on stream->get_id(), and I didn't want to change that everywhere otherwise it would be a much bigger change.  get_name and get_id are virtual methods in the base class, but the base class doesn't know about the class statics, so it cannot call get_id_static(). I just followed the same model already used for get_name() which calls get_name_static(). You will see a get_name() implementation in each subclass that calls the static.